specs/networking/p2p-interface.md

```diff
@@ -0,0 +1,712 @@
+
+Connection-level and stream-level (see the rationale section below for explanations) protocol negotiation MUST be conducted using [multistream-select v1.0](https://github.com/multiformats/multistream-select/). Its protocol ID is: `/multistream/1.0.0`.
+
+#### Mainnet
+
+Clients MUST support [multistream-select 1.0](https://github.com/multiformats/multistream-select/) and MAY support [multiselect 2.0](https://github.com/libp2p/specs/pull/95). Depending on the number of clients that have implementations for multiselect 2.0 by mainnet, [multistream-select 1.0](https://github.com/multiformats/multistream-select/) may be phased out.
+
+## Multiplexing
+
+During connection bootstrapping, libp2p dynamically negotiates a mutually supported multiplexing method to conduct parallel conversations. This applies to transports that are natively incapable of multiplexing (e.g. TCP, WebSockets, WebRTC), and is omitted for capable transports (e.g. QUIC).
+
+Two multiplexers are commonplace in libp2p implementations: [mplex](https://github.com/libp2p/specs/tree/master/mplex) and [yamux](https://github.com/hashicorp/yamux/blob/master/spec.md). Their protocol IDs are, respectively: `/mplex/6.7.0` and `/yamux/1.0.0`.
+
+Clients MUST support [mplex](https://github.com/libp2p/specs/tree/master/mplex) and MAY support [yamux](https://github.com/hashicorp/yamux/blob/master/spec.md). If both are supported by the client, yamux must take precedence during negotiation. See the Rationale section of this document for tradeoffs.
+
+# ETH2 network interaction domains
+
+## Constants
+
+This section outlines constants that are used in this spec.
+
+- `RQRP_MAX_SIZE`: The max size of uncompressed req/resp messages that clients will allow.
+ Value: TBD
+- `GOSSIP_MAX_SIZE`: The max size of uncompressed gossip messages
+ Value: 1MB (estimated from expected largest uncompressed block size).
+- `SHARD_SUBNET_COUNT`: The number of shard subnets used in the gossipsub protocol.
+ Value: TBD
+
+## The gossip domain: gossipsub
+
+Clients MUST support the [gossipsub](https://github.com/libp2p/specs/tree/master/pubsub/gossipsub) libp2p protocol.
+
+**Protocol ID:** `/meshsub/1.0.0`
+
+**Gossipsub Parameters**
+
+*Note: Parameters listed here are subject to a large-scale network feasibility study.*
+
+The following gossipsub parameters will be used:
+
+- `D` (topic stable mesh target count): 6
+- `D_low` (topic stable mesh low watermark): 4
+- `D_high` (topic stable mesh high watermark): 12
+- `D_lazy` (gossip target): 6
+- `fanout_ttl` (ttl for fanout maps for topics we are not subscribed to but have published to, seconds): 60
+- `gossip_advertise` (number of windows to gossip about): 3
+- `gossip_history` (number of heartbeat intervals to retain message IDs): 5
+- `heartbeat_interval` (frequency of heartbeat, seconds): 1
+
+### Topics
+
+Topics are plain UTF-8 strings, and are encoded on the wire as determined by protobuf (gossipsub messages are enveloped in protobuf messages).
```

Comment:
Is there a rationale for storing topics as plain UTF-8 strings instead of topic hashes?

Reply:
Good point. Yes. I will add the response to the rationale section and crosslink the commit here.

Reply:
Actually, it was already there. But I took the opportunity to enhance the answer anyway: https://github.com/status-im/eth2.0-specs/commit/1ba03b4c33a6ade51dbd60dae9a6a15acd7be531

Reply:
For keeping the comment reel tidy, this thread can be marked as resolved @arnetheduck (I can't do it).

Reply:
breathe in, breathe out :)